shared/src/native-src/loader.cpp

```diff
@@ -322,52 +328,97 @@ namespace shared
                       ToString(_corlibMetadata.Metadata.usBuildNumber) +
                       "]");
             }
+        }
+        
+        if (_loaderOptions.RewriteMSCorLibMethods && _loaderOptions.IsNet46OrGreater && _corlibMetadata.Token != mdAssemblyNil)
+        {
+            mdAssemblyRef corlibAssemblyRef = mdAssemblyRefNil;
+            mdTypeRef systemObjectTypeRef = mdTypeRefNil;
+            mdTypeDef loaderTypeDef = mdTypeDefNil;
+            mdMethodDef loaderMethodDef = mdMethodDefNil;
+            mdMemberRef securitySafeCriticalCtorMemberRef = mdMemberRefNil;
 
-            if (_loaderOptions.RewriteMSCorLibMethods && _loaderOptions.IsNet46OrGreater)
+            for (const auto& specMethodPair : _specificMethodsToRewrite)
             {
+                // Check assembly name
+                const auto& specificAssemblyInjectName = std::get<0>(specMethodPair);
+                if (specificAssemblyInjectName != assemblyNameString)
+                {
+                    continue;
+                }
+
+                const auto& specificTypeToInjectName = std::get<1>(specMethodPair);
+                const auto& specificMethodToInjectName = std::get<2>(specMethodPair);
+
                 mdTypeDef appDomainTypeDef;
```

Comment:
nit: I think this variable could be renamed now

Reply:
oh you're right, I will change it.

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/2504